student/src/main/java/tds/student/sql/repository/RemoteExamRepository.java

```diff
@@ -0,0 +1,50 @@
+package tds.student.sql.repository;
+
+import TDS.Shared.Exceptions.ReturnStatusException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.client.RestClientException;
+import org.springframework.web.client.RestTemplate;
+
+import tds.common.Response;
+import tds.exam.Exam;
+import tds.exam.OpenExamRequest;
+import tds.student.sql.abstractions.ExamRepository;
+
+@Repository
+public class RemoteExamRepository implements ExamRepository {
```

Comment:
I go back and forth whether we need a repository but I feel it is cleaner to have our data access be here rather than have the service directly make the http calls.